src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -27,4 +27,30 @@ public Integer getLength() {
     public String getColor() {
         return color;
     }
+
+    @Override
+    public boolean equals(Object object) {
+        if (this == object) {
+            return true;
+        }
+        if (object == null || !object.getClass().equals(Rectangle.class)) {
+            return false;
+        }
+        Rectangle rectangle = (Rectangle) object;
+        return (width == null && rectangle.width == null)
+                || (width != null && width.equals(rectangle.width))
+                && (length == null && rectangle.length == null)
+                || (length != null && length.equals(rectangle.length))
+                && (color == null && rectangle.color == null)
+                || (color != null && color.equals(rectangle.color));
```

Comment:
The logical-AND operator ( && ) has higher precedence than the logical-OR operator ( || )
add parenthesis to this statement to ensure the flow is correct

Reply:
done